### DISCUSSION TITLE
rename skip to precondition

### DISCUSSION THREAD

**Discussion by fokion** [ovh/venom#696] [Ideas]
In order to skip a test case or a step the assertions of the skip block **MUST** be false and when we have all the assertions valid under the skip block then it will execute it. This is confusing in my opinion as it is acting as a precondition of our assertions. 
I propose to rename this block to _**precondition**_ and if we want to use the **skip** field we can convert it so that if everything is asserted as true then it must actually skip it.

**Comment by ivan-velasco** [comment]
This brings up the question, should there be support in test suites for before and after routines also? At present, I find myself needing to craft a test prior to and following an entire set of tests for setups and teardown routines. You get additional "tests" counted in the final result which can be misleading. 
For instance:
Original:
```
test: get_login_token
test: testcases_to_run
test: remove_tenants
```

After:
```
before: get_login_token
test: testcases_to_run
after: remove_tenants
```

**Comment by yesnault** [comment]
I'm agree that it can be tricky to well understand the condition skip.

For example:

```yml
name: "Skip testsuite"
vars:
  foo: bar

testcases:
- name: testA
  skip: 
    - foo ShouldEqual aaa
  steps:
  - type: exec
    script: echo "it's skipped"
 ```
returns:
```
❯ venom run -vv a.yml
          [trac] writing venom.49.log
 • Skip testsuite (a.yml)
        • testA
                • (all steps were skipped)
final status: SKIP
```
 
 say that: 
- "if one condition from the skip block is not true, it's skipped"

Adding https://github.com/ovh/venom/pull/806 for now.
